.travis.yml

```diff
@@ -13,20 +13,18 @@ branches:
 env:
   # add target platforms to build matrix
   # lint is a target so we don't need to lint multiple times
-  - TARGET='build_all PLATFORM=x86'
+  - TARGET='build_all PLATFORM=x86 CC=gcc-6 LD=gcc-6'
   - TARGET='build_all PLATFORM=stm32f0xx'
-  - TARGET='test_all PLATFORM=x86'
+  - TARGET='test_all PLATFORM=x86 CC=gcc-6 LD=gcc-6'
   - TARGET=lint
 
-before_install:
-  # update the libraries (ms-common, etc.)
-  - git submodule update --init --recursive
-
 before_script:
-  # install GCC ARM from GNU ARM Embedded Toolchain PPA
+  # install GCC ARM from GNU ARM Embedded Toolchain PPA and GCC 6
   - sudo add-apt-repository ppa:team-gcc-arm-embedded/ppa -y
+  - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
```

Comment:
Okay. Apparently it's supposed to be faster to do it this way? I can change it though.
https://gist.github.com/cotsog/3ce84675af0d74438d91